### PR TITLE
feat(settings): disable `show accounts` when multiple accounts are authenticated

### PR DESCRIPTION
**Pull request by setchy**
Toggling this setting has no affect when there are multiple accounts authenticated.

### CODE REVIEW

File: src/utils/auth/utils.ts [hunk 1]
```diff
@@ -218,3 +218,11 @@ export function isValidToken(token: Token) {
 export function getAccountUUID(account: Account): string {
   return btoa(`${account.hostname}-${account.user.id}-${account.method}`);
 }
+
+export function hasAccounts(auth: AuthState) {
```
Comment (afonsojramos) [inline src/utils/auth/utils.ts:222]: 😅

File: src/components/settings/AppearanceSettings.tsx [hunk 2]
```diff
@@ -198,7 +199,8 @@ export const AppearanceSettings: FC = () => {
       <Checkbox
         name="showAccountHostname"
         label="Show account hostname"
-        checked={settings.showAccountHostname}
+        checked={settings.showAccountHostname || hasMultipleAccounts(auth)}
+        disabled={hasMultipleAccounts(auth)}
```
Comment (afonsojramos) [inline src/components/settings/AppearanceSettings.tsx:203]: Why exactly do we need to disable this on multiple accounts?

Comment (afonsojramos) [reply src/components/settings/AppearanceSettings.tsx:203]: I see, but can't the hostname be the same on some situations?

Comment (setchy) [reply src/components/settings/AppearanceSettings.tsx:203]: Indeed, we support repeated hostnames now.

Comment (setchy) [reply src/components/settings/AppearanceSettings.tsx:203]: so single account, feature flag toggles showing the account header
multiple accounts, feature flag does nada

Comment (afonsojramos) [reply src/components/settings/AppearanceSettings.tsx:203]: Right, but does it make sense for it to be disabled on the situations where it is the same for example?

Comment (setchy) [reply src/components/settings/AppearanceSettings.tsx:203]: `showAccountHostname` is misleading nowdays.  It really should be `showAccountNameHeader` or something like that as it represents the row containing account name, account avatar, account platform type, etc

**Pull request merged**